packages/protocol/contracts/tokenvault/ERC721Vault.sol

```diff
@@ -134,16 +130,9 @@ contract ERC721Vault is BaseNFTVault, IERC721ReceiverUpgradeable {
     {
         checkRecallMessageContext();
 
-        if (message.owner == address(0)) revert VAULT_INVALID_USER();
-        if (message.srcChainId != block.chainid) {
-            revert VAULT_INVALID_SRC_CHAIN_ID();
-        }
-
```

Comment:
These types of checks are also only done here in recall for ERC721 tokens, but not in any of the other vaults. So if they are necessary, which I don't think is the case, but if they are they need to be done in other vaults as well.

Reply:
I think we should add `if (message.srcChainId != block.chainid)` check to the other two vaults as well.

Reply:
I think none of these checks (as pointed out and deleted by Brecht is not necessary).
`message.owner == address(0)` -> Already checked in sendMessage so it does not get till the recall. If a message is maliciously constructed, proving already throw.
`message.srcChainId != block.chainid` -> Also alredy checked in sendMessage, and in any case a message is maliciously constructed, it cannot make through a proof check an also a valid context check in `checkRecallMessageContext` because it could not even go through the `sendMessage`.

Reply:
Fine with me.